### PR TITLE
 Ansible linter

### PR DESCRIPTION
**Pull request by lexsys27**
Testing change to deployment playbooks includes syntax check, checking the code follows best practices and integration tests. This PR addresses checking code for following best practices using [ansible lint](https://github.com/willthames/ansible-lint) tool. To list the rules run `ansible-lint -L`.

Each time changes are commited git runs pre-commit hook ( db944c8 ). This hook runs `ansible-lint` and allows to commit changes only if there are no linter errors.

As there is not local development environment available (see #75 ) tests will be done manually. Each node is provisioned on Azure. If provisioning succeeds the test is passed. Else - test failed.

Ansible playbooks that provision EC2 resources are out of scope for this PR. Hook just skips their checks.

---

- [X] Modify playbooks to pass `ansible-lint` tests [#82] 
  - [X] Make package installs repeatable 
  - [X] Add tasks to generate SSL certificates 
  - [X] Mute warning on swap tasks
  - [X] Clone specific version of the codebase from the GitHub [#84]
  - [X] Use copy module instead of shell command
  - [X] Configure task to install npm modules
  - [X] Name the unnamed tasks in Ansible playbooks
  - [X] Add pre-commit hook to check playbook syntax
  - [X] Fix linter errors on netstat role
- [x] Verify the changes 
  - [x] netstat
     - [x] fix permission problems with `usermanager` role [#83]
  - [x] moc
  - [x] bootnode
  - [x] validator
  - [x] explorer
  - [x] fix discovered issues
    - [x] fix `ansible-lint` trailing whitespace warning  in EC2 playbooks
    - [x] fix include deprecation warning in the "*-access" roles

### CODE REVIEW

File: hooks/pre-commit.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# To install run
+#   ln -s ../../hooks/pre-commit.sh .git/hooks/pre-commit
+
+echo "Running pre-commit hook"
+
+STASH_NAME="pre-commit-$(date +%s)"
+git stash save -q --keep-index $STASH_NAME
+
+echo "Checking the playbook syntax"
+ansible-playbook -i hosts site.yml --syntax-check
```
Comment (phahulin) [inline hooks/pre-commit.sh]: Does it actually need `hosts`? Maybe `-i none` will do?

File: roles/explorer/tasks/main.yml [hunk 2]
```diff
@@ -26,17 +26,23 @@
   with_items:
    - node.toml
 
-- git: repo=https://github.com/{{ MAIN_REPO_FETCH }}/chain-explorer dest={{ home }}/chain-explorer
+- name: Clone poanetworks/chain-explorer repo from GitHub
+  git:
+    repo: "https://github.com/{{ MAIN_REPO_FETCH }}/chain-explorer"
+    dest: "{{ home }}/chain-explorer"
+    version: "acee07c"
```
Comment (phahulin) [inline roles/explorer/tasks/main.yml]: Maybe better download latest version?

File: roles/netstat/tasks/main.yml [hunk 3]
```diff
@@ -7,27 +7,39 @@
     owner: "{{ username }}"
     group: "{{ username }}"
 
-- git: repo=https://github.com/{{ MAIN_REPO_FETCH }}/eth-netstats dest={{ home }}/eth-netstats
+- name: Clone poanetworks/eth-netstats repo from GitHub
+  git:
+    repo: "https://github.com/{{ MAIN_REPO_FETCH }}/eth-netstats"
+    dest: "{{ home }}/eth-netstats"
+    version: "cb431d6"
```
Comment (phahulin) [inline roles/netstat/tasks/main.yml]: Also here - maybe better download latest version?

File: roles/poa-netstats/tasks/main.yml [hunk 4]
```diff
@@ -15,17 +15,23 @@
     owner: "{{ username }}"
     group: "{{ username }}"
 
-- git: repo=https://github.com/{{ MAIN_REPO_FETCH }}/eth-net-intelligence-api dest={{ home }}/eth-net-intelligence-api
+- name: Clone poanetworks/eth-net-intelligence-api repo from GitHub
+  git:
+    repo: "https://github.com/{{ MAIN_REPO_FETCH }}/eth-net-intelligence-api"
+    dest: "{{ home }}/eth-net-intelligence-api"
+    version: "9773b5b"
```
Comment (phahulin) [inline roles/poa-netstats/tasks/main.yml]: Also here about latest version

File: roles/explorer/tasks/main.yml (hunk 2 above)
Comment (lexsys27) [reply roles/explorer/tasks/main.yml]: I have moved the discussion to #84

File: hooks/pre-commit.sh (hunk 1 above)
Comment (lexsys27) [reply hooks/pre-commit.sh]: fixed in e38f34c 

It produces a lot of warnings, though. Looks like Ansible have no option to suppress these warnings

**Comment by lexsys27** [comment]
@phahulin are there any issues left preventing this PR from merging?

**Pull request merged**